app/controllers/internal/events_controller.rb

```diff
@@ -14,8 +18,75 @@ def show
       @page_title = @event.name
     end
 
+    def new
+      @event = Event.new(venue_type: Event::VENUE_TYPES[:existing])
+      @event.building = EventBuilding.new
+    end
+
+    def approve
+      api_event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:format])
+      api_event.status_id = GetIntoTeachingApiClient::Constants::EVENT_STATUS["Open"]
+      submit_to_api(api_event)
+      redirect_to internal_events_path(success: true)
+    end
+
+    def create
+      @event = Event.new(event_params)
+      @event.building = format_building(event_params)
+      if submit_pending
+        redirect_to internal_events_path(success: :pending)
+      else
+        render action: :new
+      end
+    end
+
+    def edit
+      api_event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+      @event = Event.initialize_with_api_event(api_event)
+      render :new
+    end
+
   private
 
+    def submit_pending
+      return if invalid?
+
+      @event.status_id = GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
+      api_event = @event.to_api_event
+      submit_to_api(api_event)
+    end
+
+    def invalid?
+      invalid_building = @event.building.nil? ? false : @event.building.invalid?
```

Comment:
Can this not be encapsulated in the `Internal::Event` class so you can just call `@event.invalid?`?